packages/core/strapi/lib/Strapi.js

```diff
@@ -72,7 +72,9 @@ const resolveWorkingDirectories = (opts) => {
 /** @implements {import('@strapi/strapi').Strapi} */
 class Strapi {
   constructor(opts = {}) {
-    destroyOnSignal(this);
+    if (opts.destroyOnSignal !== false) {
+      destroyOnSignal(this);
+    }
```

Comment:
@alexandrebodin I need a way to (optionally) create an instance of Strapi without having it call `process.exit()` when it receives a SIGINT, so that we can gracefully abort a transfer and then destroy the instance of Strapi and then exit.

This was the first thing that came to mind, but I wanted to get your opinion before adding new options to the Strapi constructor.